src/main/java/frc/robot/utils/MovingAverage.java

```diff
@@ -0,0 +1,114 @@
+package frc.robot.utils;
+
+import java.io.*;
+import java.util.Map;
+import java.util.TreeMap;
+
+/**
+ * The moving average contains a list of velocity based on the distance, and should give an approximation of the velocity to apply.
+ * In order to the moving average object to approximate better, you need to pass more values.
+ * The object calculate calculate a straight line between the closest point, and retrieve the velocity (in the straight line between the records).
+ */
+public class MovingAverage {
+    private final String DELIMITER = ",";
+    private final Map<Double, Double> distanceVelocityMap; // <key - distance, value - velocity>, tree map is automatically sort the values by distance.
+
+    public MovingAverage(String pathToCsv) {
+        this(new File(pathToCsv));
+    }
+
+    public MovingAverage(File csv) {
+        this.distanceVelocityMap = new TreeMap<>();
+        if (csv.exists() && csv.isFile()) {
```

Comment:
```suggestion
        if (csv.isFile()) {
```
[Hint](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#isFile()).

Reply:
Aren't you going to remove the redundant check?